File: backend/app/schemas/nutrition.py
from pydantic import BaseModel



class NutritionRequest(BaseModel):

    sport: str

    distance: float

    elevation: float

    duration: float

    temperature: float

    weight: float

    intensity: str




class NutritionRequirementsResponse(BaseModel):

    carbs_per_hour: int

    water_per_hour: int

    sodium_per_hour: int




class NutritionEventResponse(BaseModel):

    time: int

    type: str

    carbs: int




class NutritionResponse(BaseModel):

    activity: dict

    requirements: NutritionRequirementsResponse

    plan: list[NutritionEventResponse]

File: frontend/src/types/nutrition.ts
export type Sport =
    | "trail"
    | "cycling"
    | "hiking"
    | "mtb";


export type Intensity =
    | "low"
    | "moderate"
    | "high";

export interface NutritionRequest {

    sport: Sport;

    distance: number;

    elevation: number;

    duration: number;

    temperature: number;

    weight: number;

    intensity: Intensity;

}


export interface NutritionRequirements {

    carbs_per_hour: number;

    water_per_hour: number;

    sodium_per_hour: number;

}


export interface NutritionEvent {

    time:number;

    type:string;

    carbs:number;

}


export interface NutritionResponse {

    activity: {

        sport: string;

        distance: number;

        duration: number;

    };


    requirements: NutritionRequirements;


    plan: NutritionEvent[];

}